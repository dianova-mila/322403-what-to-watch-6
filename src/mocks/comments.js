export default [
  {
    "id": 1,
    "user": {
      "id": 10,
      "name": `Max`
    },
    "rating": 3.9,
    "comment": `Poised to be an instant classic, almost everything about this film is phenomenal - the acting, the cinematography, the discography, etc.`,
    "date": `2021-02-18T08:04:28.658Z`
  },
  {
    "id": 2,
    "user": {
      "id": 19,
      "name": `Christina`
    },
    "rating": 9.7,
    "comment": `It was well acted, directed, and the music was good. But the story is yawn. Not trying to rip anybody but I checked my watch a dozen times during this movie.`,
    "date": `2021-02-17T08:04:28.658Z`
  }
];
