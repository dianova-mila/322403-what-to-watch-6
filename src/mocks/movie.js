export default {
  name: `Macbeth`,
  posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Macbeth.jpg`,
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/macbeth.jpg`,
  backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Macbeth.jpg`,
  backgroundColor: `#F1E9CE`,
  description: `Macbeth, the Thane of Glamis, receives a prophecy from a trio of witches that one day he will become King of Scotland. Consumed by ambition and spurred to action by his wife, Macbeth murders his king and takes the throne for himself.`,
  rating: 3.3,
  scoresCount: 48798,
  director: `Justin Kurzel`,
  starring: [
    `Michael Fassbender`,
    `Marion Cotillard`,
    `Jack Madigan`
  ],
  runTime: 113,
  genre: `Drama`,
  released: 2015,
  id: `1`,
  isFavorite: false,
  videoLink: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
};
