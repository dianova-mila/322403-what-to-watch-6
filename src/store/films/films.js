import {createReducer} from "@reduxjs/toolkit";
import {changeGenre, loadMovies, getMovieList, loadFavorites} from "../actions";
import {ALL_GENRES} from "../../const";

const initialState = {
  currentGenre: ALL_GENRES,
  movieList: [],
  movies: [],
  isDataLoaded: false,
  favoritesMovies: [],
  isFavoritesLoaded: false
};

const films = createReducer(initialState, (builder) => {
  builder.addCase(changeGenre, (state, action) => {
    state.currentGenre = action.payload;
  });

  builder.addCase(loadMovies, (state, action) => {
    state.movies = action.payload;
    state.movieList = action.payload;
    state.isDataLoaded = true;
  });

  builder.addCase(getMovieList, (state, action) => {
    state.movieList = action.payload;
  });

  builder.addCase(loadFavorites, (state, action) => {
    state.favoritesMovies = action.payload;
    state.isFavoritesLoaded = true;
  });
});

export {films};
