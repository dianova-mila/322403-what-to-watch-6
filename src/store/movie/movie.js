import {createReducer} from "@reduxjs/toolkit";
import {loadOneMovie, loadPromoMovie, loadComments} from "../actions";

const initialState = {
  movie: {},
  promoMovie: {},
  isPromoMovieLoaded: false,
  comments: [],
  isOneMovieLoaded: false,
};

const movie = createReducer(initialState, (builder) => {
  builder.addCase(loadOneMovie, (state, action) => {
    state.movie = action.payload;
    state.isOneMovieLoaded = true;
  });

  builder.addCase(loadPromoMovie, (state, action) => {
    state.promoMovie = action.payload;
    state.isPromoMovieLoaded = true;
  });

  builder.addCase(loadComments, (state, action) => {
    state.comments = action.payload;
  });
});

export {movie};
