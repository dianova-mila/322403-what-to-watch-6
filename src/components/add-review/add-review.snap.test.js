import React from "react";
import {Router} from "react-router-dom";
import {render} from "@testing-library/react";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import {createMemoryHistory} from "history";
import movie from "../../mocks/movie";
import films from "../../mocks/films";
import comments from "../../mocks/comments";
import App from "../app/app";

const mockStore = configureStore({});

it(`AddReview should render correctly`, () => {
  const store = mockStore({
    USER: {authorizationStatus: true, userInfo: {avatarUrl: `img/1.png`}},
    FILMS: {movies: films},
    MOVIE: {movie, comments, isOneMovieLoaded: true}
  });

  const history = createMemoryHistory();
  history.push(`/films/1/review`);

  const {container} = render(
      <Provider store={store}>
        <Router history={history}>
          <App />
        </Router>
      </Provider>
  );

  expect(container).toMatchSnapshot();
});
