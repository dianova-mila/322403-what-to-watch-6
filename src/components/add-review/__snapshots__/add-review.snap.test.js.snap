// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddReview should render correctly 1`] = `
<div>
  <section
    class="movie-card movie-card--full"
    style="background-color: rgb(241, 233, 206);"
  >
    <div
      class="movie-card__hero"
    >
      <div
        class="movie-card__bg"
      >
        <img
          alt="Macbeth"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Macbeth.jpg"
        />
      </div>
      <h1
        class="visually-hidden"
      >
        WTW
      </h1>
      <header
        class="page-header movie-card__head"
      >
        <div
          class="logo"
        >
          <a
            class="logo__link"
            href="/"
          >
            <span
              class="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              class="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              class="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          class="user-block"
        >
          <div
            class="user-block__avatar"
          >
            <img
              alt="User avatar"
              height="63"
              src="img/1.png"
              width="63"
            />
          </div>
        </div>
      </header>
      <div
        class="movie-card__wrap"
      >
        <div
          class="movie-card__desc"
        >
          <h2
            class="movie-card__title"
          >
            Macbeth
          </h2>
          <p
            class="movie-card__meta"
          >
            <span
              class="movie-card__genre"
            >
              Drama
            </span>
            <span
              class="movie-card__year"
            >
              2015
            </span>
          </p>
          <div
            class="movie-card__buttons"
          >
            <button
              class="btn btn--play movie-card__button"
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlink:href="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              class="btn btn--list movie-card__button"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlink:href="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
            <a
              class="btn movie-card__button"
              href="/films/1/review"
            >
              Add review
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="movie-card__wrap movie-card__translate-top"
    >
      <div
        class="movie-card__info"
      >
        <div
          class="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="Macbeth"
            height="327"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Macbeth.jpg"
            width="218"
          />
        </div>
        <div
          class="movie-card__desc"
        >
          <nav
            class="movie-nav movie-card__nav"
          >
            <ul
              class="movie-nav__list"
            >
              <li
                class="movie-nav__item "
              >
                <a
                  class="movie-nav__link"
                  href="#"
                >
                  Overview
                </a>
              </li>
              <li
                class="movie-nav__item "
              >
                <a
                  class="movie-nav__link"
                  href="#"
                >
                  Details
                </a>
              </li>
              <li
                class="movie-nav__item movie-nav__item--active"
              >
                <a
                  class="movie-nav__link"
                  href="#"
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
          <div
            class="add-review"
          >
            
            <form
              action="#"
              class="add-review__form"
            >
              <div
                class="rating"
              >
                <div
                  class="rating__stars"
                >
                  <input
                    class="rating__input"
                    id="star-1"
                    name="rating"
                    type="radio"
                    value="1"
                  />
                  <label
                    class="rating__label"
                    for="star-1"
                  >
                    Rating 1
                  </label>
                  <input
                    class="rating__input"
                    id="star-2"
                    name="rating"
                    type="radio"
                    value="2"
                  />
                  <label
                    class="rating__label"
                    for="star-2"
                  >
                    Rating 2
                  </label>
                  <input
                    class="rating__input"
                    id="star-3"
                    name="rating"
                    type="radio"
                    value="3"
                  />
                  <label
                    class="rating__label"
                    for="star-3"
                  >
                    Rating 3
                  </label>
                  <input
                    class="rating__input"
                    id="star-4"
                    name="rating"
                    type="radio"
                    value="4"
                  />
                  <label
                    class="rating__label"
                    for="star-4"
                  >
                    Rating 4
                  </label>
                  <input
                    class="rating__input"
                    id="star-5"
                    name="rating"
                    type="radio"
                    value="5"
                  />
                  <label
                    class="rating__label"
                    for="star-5"
                  >
                    Rating 5
                  </label>
                  <input
                    class="rating__input"
                    id="star-6"
                    name="rating"
                    type="radio"
                    value="6"
                  />
                  <label
                    class="rating__label"
                    for="star-6"
                  >
                    Rating 6
                  </label>
                  <input
                    class="rating__input"
                    id="star-7"
                    name="rating"
                    type="radio"
                    value="7"
                  />
                  <label
                    class="rating__label"
                    for="star-7"
                  >
                    Rating 7
                  </label>
                  <input
                    class="rating__input"
                    id="star-8"
                    name="rating"
                    type="radio"
                    value="8"
                  />
                  <label
                    class="rating__label"
                    for="star-8"
                  >
                    Rating 8
                  </label>
                  <input
                    class="rating__input"
                    id="star-9"
                    name="rating"
                    type="radio"
                    value="9"
                  />
                  <label
                    class="rating__label"
                    for="star-9"
                  >
                    Rating 9
                  </label>
                  <input
                    class="rating__input"
                    id="star-10"
                    name="rating"
                    type="radio"
                    value="10"
                  />
                  <label
                    class="rating__label"
                    for="star-10"
                  >
                    Rating 10
                  </label>
                </div>
              </div>
              <div
                class="add-review__text"
              >
                <textarea
                  class="add-review__textarea"
                  id="review-text"
                  maxlength="400"
                  minlength="50"
                  name="review-text"
                  placeholder="Review text"
                  required=""
                />
                <div
                  class="add-review__submit"
                >
                  <button
                    class="add-review__btn"
                    disabled=""
                    type="submit"
                  >
                    Post
                  </button>
                </div>
              </div>
            </form>
          </div>
          <div
            class="movie-card__reviews movie-card__row"
          >
            <div
              class="movie-card__reviews-col"
            >
              <div
                class="review"
              >
                <blockquote
                  class="review__quote"
                >
                  <p
                    class="review__text"
                  >
                    Poised to be an instant classic, almost everything about this film is phenomenal - the acting, the cinematography, the discography, etc.
                  </p>
                  <footer
                    class="review__details"
                  >
                    <cite
                      class="review__author"
                    >
                      Max
                    </cite>
                    <time
                      class="review__date"
                      datetime="{dayjs(new Date(item.date)).format(\`YYYY-MM-DD\`)}"
                    >
                      February 18, 2021
                    </time>
                  </footer>
                </blockquote>
                <div
                  class="review__rating"
                >
                  3.9
                </div>
              </div>
              <div
                class="review"
              >
                <blockquote
                  class="review__quote"
                >
                  <p
                    class="review__text"
                  >
                    It was well acted, directed, and the music was good. But the story is yawn. Not trying to rip anybody but I checked my watch a dozen times during this movie.
                  </p>
                  <footer
                    class="review__details"
                  >
                    <cite
                      class="review__author"
                    >
                      Christina
                    </cite>
                    <time
                      class="review__date"
                      datetime="{dayjs(new Date(item.date)).format(\`YYYY-MM-DD\`)}"
                    >
                      February 17, 2021
                    </time>
                  </footer>
                </blockquote>
                <div
                  class="review__rating"
                >
                  9.7
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
  <div
    class="page-content"
  >
    <section
      class="catalog catalog--like-this"
    >
      <h2
        class="catalog__title"
      >
        More like this
      </h2>
      <div
        class="catalog__movies-list"
      >
        <article
          class="small-movie-card catalog__movies-card"
        >
          <div
            class="small-movie-card__image"
          >
            <img
              alt="Macbeth"
              height="175"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/macbeth.jpg"
              width="280"
            />
          </div>
          <h3
            class="small-movie-card__title"
          >
            <a
              class="small-movie-card__link"
              href="/films/1/"
            >
              Macbeth
            </a>
          </h3>
        </article>
        <article
          class="small-movie-card catalog__movies-card"
        >
          <div
            class="small-movie-card__image"
          >
            <img
              alt="Orlando"
              height="175"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/orlando.jpg"
              width="280"
            />
          </div>
          <h3
            class="small-movie-card__title"
          >
            <a
              class="small-movie-card__link"
              href="/films/3/"
            >
              Orlando
            </a>
          </h3>
        </article>
        <article
          class="small-movie-card catalog__movies-card"
        >
          <div
            class="small-movie-card__image"
          >
            <img
              alt="Aviator"
              height="175"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/aviator.jpg"
              width="280"
            />
          </div>
          <h3
            class="small-movie-card__title"
          >
            <a
              class="small-movie-card__link"
              href="/films/8/"
            >
              Aviator
            </a>
          </h3>
        </article>
      </div>
    </section>
    <footer
      class="page-footer"
    >
      <div
        class="logo"
      >
        <a
          class="logo__link logo__link--light"
          href="/"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        class="copyright"
      >
        <p>
          © 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>
</div>
`;
